### PR TITLE
test: fix TEST_PREVIOUS_RELEASES check

### PR DESCRIPTION
**Pull request by fanquake**
Pythons [`os.getenv()`](https://docs.python.org/3.8/library/os.html#os.getenv) returns `None` by default when a key isn't found. So if this test is run locally and you haven't set `TEST_PREVIOUS_RELEASES` or previous release binaries aren't available, it will always fail.

```bash
Remaining jobs: [wallet_upgradewallet.py]
1/1 - wallet_upgradewallet.py failed, Duration: 0 s

stdout:
2020-05-07T03:24:36.506000Z TestFramework (INFO): Initializing test directory /var/folders/z2/cn877pxd3czdfh47mfkmbwgm0000gn/T/test_runner_₿_🏃_20200507_112436/wallet_upgradewallet_0
2020-05-07T03:24:36.514000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "bitcoin/test/functional/test_framework/test_framework.py", line 471, in start_nodes
    node.start(extra_args[i], *args, **kwargs)
  File "bitcoin/test/functional/test_framework/test_node.py", line 205, in start
    self.process = subprocess.Popen(self.args + extra_args, env=subp_env, stdout=stdout, stderr=stderr, cwd=cwd, **kwargs)
  File "/Users/x/.pyenv/versions/3.5.6/lib/python3.5/subprocess.py", line 676, in __init__
    restore_signals, start_new_session)
  File "/Users/x/.pyenv/versions/3.5.6/lib/python3.5/subprocess.py", line 1289, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'bitcoin/releases/v0.16.3/bin/bitcoind'

During handling of the above exception, another exception occurred:
```

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Not sure why we are doing bash env var juggling here. This should be an option on the test framework like everything else. E.g. `--factor`

**Comment by fanquake** [comment]
> Not sure why we are doing bash env var juggling here.

🤷‍♂️  I just want to be able to run the full functional test suite locally without pointless failures. I don't understand how we've ended up with a test that will fail by default when run outside the CI.

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -672,7 +672,7 @@ def skip_if_no_previous_releases(self):
 
     def has_previous_releases(self):
         """Checks whether previous releases are present and enabled."""
-        if os.getenv("TEST_PREVIOUS_RELEASES") == "false":
+        if os.getenv("TEST_PREVIOUS_RELEASES") == "false" or os.getenv("TEST_PREVIOUS_RELEASES") is None:
```
Comment (MarcoFalke) [inline test/functional/test_framework/test_framework.py:675]:
```suggestion
        if os.getenv("TEST_PREVIOUS_RELEASES", "false") == "false":
```

does this pass?

**Comment by MarcoFalke** [comment]
A workaround could also be to remove the prev releases folder

**Comment by Sjors** [comment]
That's odd, on macOS it skips it. Maybe the difference between bash and zsh?

**Comment by fanquake** [comment]
Closing in favour of #19014.

**Comment by MarcoFalke** [comment]
Just to make sure, #1904 doesn't change any behaviour. You will still run into failures if the prev releases dir exists but is missing the binaries. The only way to fix that is to remove the dir or add the binaries.

**Pull request closed without merging**